test/integration/spec/exec.ts

```diff
@@ -0,0 +1,50 @@
+import * as chai from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
```

Comment:
We often use Jest as a testing framework because it not only does not require a separate matcher library like `chai`, but it also can do [promise matching out of the box](https://facebook.github.io/jest/docs/en/expect.html#resolves) as well as [code coverage](https://facebook.github.io/jest/docs/en/configuration.html#coveragethreshold-object). Jest's API is basically the same as Mocha's. See [`invokable` unit tests](https://github.com/kourge/invokable/blob/master/src/index.test.ts) for an example.

Reply:
fair enough; i'll do that in the next PR, this one has gotten big enough.